File: features/snippets/external_storage/multiple_drivers.py
from temporalio.converter import ExternalStorage

from temporalio.contrib.aws.s3driver import S3StorageDriver


def configure(s3_client, legacy_driver):
    # @@@SNIPSTART python-external-storage-multiple-drivers
    preferred_driver = S3StorageDriver(client=s3_client, bucket="my-bucket")
    legacy_driver = LegacyStorageDriver()

    ExternalStorage(
        drivers=[preferred_driver, legacy_driver],
        driver_selector=lambda context, payload: preferred_driver,
    )
    # @@@SNIPEND

File: features/snippets/external_storage/multiple_drivers/multiple_drivers.ts
import type { S3Client } from '@aws-sdk/client-s3';
import { ExternalStorage, type StorageDriver } from '@temporalio/common';
import { S3StorageDriver } from '@temporalio/external-storage-s3';
import { AwsSdkS3StorageDriverClient } from '@temporalio/external-storage-s3-aws-sdk';
/* eslint-disable @typescript-eslint/no-unused-vars */

declare const s3Client: S3Client;
declare const LegacyStorageDriver: new () => StorageDriver;

{
  // @@@SNIPSTART typescript-external-storage-multiple-drivers
  const preferredDriver = new S3StorageDriver({
    client: new AwsSdkS3StorageDriverClient(s3Client),
    bucket: 'my-bucket',
  });
  const legacyDriver = new LegacyStorageDriver();

  const externalStorage = new ExternalStorage({
    drivers: [preferredDriver, legacyDriver],
    driverSelector: () => preferredDriver,
  });
  // @@@SNIPEND
}
